src/System.Private.CoreLib/src/System/Diagnostics/Eventing/EventPipeController.cs

```diff
@@ -377,26 +376,38 @@ private static void SetProviderConfiguration(string strConfig, EventPipeConfigur
         #region Configuration
 
         // Cache for COMPlus configuration variables.
-        private static int s_Config_EnableEventPipe = -1;
-        private static string s_Config_EventPipeConfig = null;
-        private static uint s_Config_EventPipeCircularMB = 0;
-        private static string s_Config_EventPipeOutputPath = null;
+        private static int s_Config_EnableEventPipe;
+        private static string s_Config_EventPipeConfig;
+        private static uint s_Config_EventPipeCircularMB;
+        private static string s_Config_EventPipeOutputPath;
 
         private static int Config_EnableEventPipe
         {
             get
             {
-                if (s_Config_EnableEventPipe == -1)
-                {
-                    string strEnabledValue = CompatibilitySwitch.GetValueInternal("EnableEventPipe");
-                    if ((strEnabledValue == null) || (!int.TryParse(strEnabledValue, out s_Config_EnableEventPipe)))
-                    {
-                        s_Config_EnableEventPipe = 0;
-                    }
-                }
+                // 1 disabled and 2 is enabled, however we return 0 and 1 respectively, so subtract 1.
+                int config_EnableEventPipe = s_Config_EnableEventPipe - 1;
+                return config_EnableEventPipe >= 0 ? config_EnableEventPipe : GetAndSetConfig_EnableEventPipe();
```

Comment:
This value is used exactly once during startup. The elaborate caching scheme is not worth it.

In fact, all the caches for the COMPlus configuration look pretty useless - all the values are used just once during startup.